uber/site_sections/dept_checklist.py

```diff
@@ -3,77 +3,135 @@
+        department_id = Department.to_id(department_id)
+        department = session.query(Department).options(
+            subqueryload(Department.dept_checklist_items)).get(department_id)
         conf = DeptChecklistConf.instances[slug]
-        if not conf.completed(attendee):
-            session.add(DeptChecklistItem(attendee=attendee, slug=slug))
-        raise HTTPRedirect('index?message={}', 'Checklist item marked as complete')
+        if department.checklist_item_for_slug(slug):
+            message = 'Checklist item already marked as complete'
+        else:
+            item = DeptChecklistItem(
+                attendee=attendee, department=department, slug=slug)
+            message = check(item)
+            if not message:
+                session.add(item)
+                message = 'Checklist item marked as complete'
+        raise HTTPRedirect(
+            'index?department_id={}&message={}', department_id, message)
 
-    def form(self, session, slug, csrf_token=None, comments=None):
+    def form(self, session, slug, department_id, csrf_token=None, comments=None):
         attendee = session.admin_attendee()
+        department_id = Department.to_id(department_id)
+        department = session.query(Department).options(
+            subqueryload(Department.dept_checklist_items)).get(department_id)
+
         conf = DeptChecklistConf.instances[slug]
-        try:
-            [item] = [item for item in attendee.dept_checklist_items if item.slug == slug]
-        except:
-            item = DeptChecklistItem(slug=slug, attendee=attendee)
+        item = department.checklist_item_for_slug(slug)
+        if not item:
+            item = DeptChecklistItem(
+                attendee=attendee, department=department, slug=slug)
 
         if comments is not None:
-            check_csrf(csrf_token)  # since this form doesn't use our normal utility methods, we need to do this manually
+            # since this form doesn't use our normal utility methods, we need
+            # to check the csrf_token manually
+            check_csrf(csrf_token)
             item.comments = comments
-            session.add(item)
-            raise HTTPRedirect('index?message={}', conf.name + ' checklist data uploaded')
+            message = check(item)
+            if not message:
+                session.add(item)
+                message = conf.name + ' checklist data uploaded'
+            raise HTTPRedirect(
+                'index?department_id={}&message={}', department_id, message)
 
         return {
             'item': item,
-            'conf': conf
+            'conf': conf,
+            'department': department
         }
 
-    def overview(self, session, message=''):
+    def overview(self, session, filtered=False, message=''):
         checklist = list(DeptChecklistConf.instances.values())
+        attendee = session.admin_attendee()
+
+        dept_filter = [Department.members_who_can_admin_checklist.any(
```

Comment:
If you go to the index without any department IDs, and you can admin multiple checklists, it brings you to the overview page with a filter showing only the departments you can admin? That's awesome!!